tests/stdlib/tasynchttpserver_transferencoding.nim

```diff
@@ -0,0 +1,75 @@
+import httpclient, asynchttpserver, asyncdispatch, asyncfutures
+import net
+
+import std/asyncnet
+import std/nativesockets
+
+const postBegin = """
+POST / HTTP/1.1
+Transfer-Encoding:chunked
+
+"""
+
+template genTest(input, expected) =
+  var sanity = false
+  proc handler(request: Request) {.async.} =
+      doAssert(request.body == expected)
+      doAssert(request.headers.hasKey("Transfer-Encoding"))
+      doAssert(not request.headers.hasKey("Content-Length"))
+      sanity = true
+      await request.respond(Http200, "Good")
+
+  let server = newAsyncHttpServer()
+  discard server.serve(Port(0), handler)
```

Comment:
(previous comment got lost)
discarding a future seems buggy (eg refs https://github.com/nim-lang/Nim/issues/11912)

note that https://github.com/nim-lang/Nim/pull/16695 doesn't have this issue (see its code with `await server.serve(Port(5555), cb)`), can you make something like it work? if not, add a comment:
```
# xxx figure out whether there's a better way than discarding a Future here
```

Reply:
So, I spent some time and dug through some docs, and with a bit of intuition and a lot of luck, I was able to replace this by using `callSoon`. I have no clue if that solution is any better than using `discard`, but I couldn't replicate what #16695 was doing. 

Please advise.

Reply:
nice! seems correct, but, optionally, also add a comment `# xxx check if there is a better way`

Reply:
It's definitely not the right thing here. All you need to do is use `asyncCheck` on the code that calls asynchttpserver's `serve`. I think you've made the code harder to work with by creating that template which is not necessary, turn it into an async proc and then `waitFor` that async proc.